stan/math/rev/functor/operands_and_partials.hpp

```diff
@@ -219,21 +223,10 @@ class ops_partials_edge<double, Op, require_eigen_st<is_var, Op>> {
  private:
   template <typename, typename, typename, typename, typename, typename>
   friend class stan::math::operands_and_partials;
-  const Op& operands_;
-
-  void dump_operands(vari** varis) {
-    Eigen::Map<promote_scalar_t<vari*, Op>>(varis, this->operands_.rows(),
-                                            this->operands_.cols())
-        = this->operands_.vi();
-  }
-  void dump_partials(double* partials) {
-    Eigen::Map<partials_t>(partials, this->partials_.rows(),
-                           this->partials_.cols())
-        = this->partials_;
-  }
-  int size() { return this->operands_.size(); }
-  std::tuple<> container_operands() { return std::tuple<>(); }
-  std::tuple<> container_partials() { return std::tuple<>(); }
+  arena_t<Op> operands_;
+  int size() const noexcept { return this->operands_.size(); }
+  inline auto operand() noexcept { return this->operands_; }
+  inline auto partial() noexcept { return this->partials_; }
```

Comment:
reference?

Reply:
Since these are arena_t types they copying is just copying the map which is cheap right?

Reply:
right